booking/src/main/java/org/dcsa/conformance/standards/booking/party/BookingCarrier.java

```diff
@@ -380,17 +380,14 @@ private String lastUrlSegment(String url) {
   }
 
   private String readCancelOperation(ConformanceRequest request) {
-    var queryParams = request.queryParams();
-    var operationParams = queryParams.get("operation");
-    if (operationParams == null || operationParams.isEmpty()) {
+    var requestPayload = request.message();
+    if (requestPayload == null || requestPayload.body() == null ) {
       return null;
     }
-    var operation = operationParams.iterator().next();
-    if (operationParams.size() > 1
-        || !(operation.equals("cancelBooking") || operation.equals("cancelAmendment"))) {
-      return "!INVALID-VALUE!";
-    }
-    return operation;
+    var cancelJsonBody = requestPayload.body().getJsonBody();
+    return cancelJsonBody.get("bookingStatus") != null ?
+      "cancelBooking" : cancelJsonBody.get("amendedBookingStatus") != null ?
+        "cancelAmendment" : "#INVALID";
```

Comment:
Please split this double `?` into multiple lines, so it is easier to read and understand it.

Reply:
Also, be sure to flag it as `#INVALID` if both attributes are present at the same time.

Reply:
FYI, The remark "Also, be sure to flag it as #INVALID if both attributes are present at the same time." remains unsolved.

Reply:
> FYI, The remark "Also, be sure to flag it as #INVALID if both attributes are present at the same time." remains unsolved.

@nt-gt  booking/src/main/java/org/dcsa/conformance/standards/booking/party/BookingCarrier.java:392 Its present here